File: drink.java
import javax.swing.*;
import javax.swing.border.EmptyBorder;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.awt.geom.RoundRectangle2D;

public class HydrationBuddy extends JFrame {
    private int intervalMinutes = 30;
    private int drinksCount = 0;
    private int timeLeft = 30;
    private boolean isSnoozed = false;
    private boolean timerRunning = false;

    private JLabel timeLeftLabel;
    private JLabel drinksLabel;
    private JLabel timerStatusLabel;
    private JButton activeIntervalBtn;

    private String[] walkingFrames = {"🚶", "🧘", "🏃", "🧘"};
    private int frameIndex = 0;

    public HydrationBuddy() {
        setTitle("Hydration Buddy Control Panel");
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setSize(500, 400);
        setLocationRelativeTo(null);
        setAlwaysOnTop(true);
        setResizable(false);

        // Main panel
        JPanel mainPanel = new JPanel();
        mainPanel.setLayout(new BorderLayout());
        mainPanel.setBackground(new Color(240, 240, 240));

        // Header
        JPanel header = createHeader();
        mainPanel.add(header, BorderLayout.NORTH);

        // Content
        JPanel content = createContent();
        JScrollPane scrollPane = new JScrollPane(content);
        scrollPane.setBorder(BorderFactory.createEmptyBorder());
        scrollPane.getViewport().setBackground(new Color(240, 240, 240));
        mainPanel.add(scrollPane, BorderLayout.CENTER);

        setContentPane(mainPanel);
        startTimer();
        setVisible(true);
    }

    private JPanel createHeader() {
        JPanel header = new JPanel();
        header.setLayout(new BoxLayout(header, BoxLayout.Y_AXIS));
        header.setBackground(new Color(102, 126, 234));
        header.setBorder(new EmptyBorder(20, 20, 20, 20));

        JLabel title = new JLabel("💧 Hydration Buddy");
        title.setFont(new Font("Helvetica", Font.BOLD, 28));
        title.setForeground(Color.WHITE);
        title.setAlignmentX(Component.CENTER_ALIGNMENT);

        JLabel subtitle = new JLabel("Stay hydrated while you code");
        subtitle.setFont(new Font("Helvetica", Font.PLAIN, 12));
        subtitle.setForeground(new Color(224, 224, 255));
        subtitle.setAlignmentX(Component.CENTER_ALIGNMENT);

        header.add(title);
        header.add(Box.createVerticalStrut(5));
        header.add(subtitle);

        return header;
    }

    private JPanel createContent() {
        JPanel content = new JPanel();
        content.setLayout(new BoxLayout(content, BoxLayout.Y_AXIS));
        content.setBackground(new Color(240, 240, 240));
        content.setBorder(new EmptyBorder(20, 20, 20, 20));

        // Interval selection
        JLabel intervalLabel = new JLabel("Reminder Interval");
        intervalLabel.setFont(new Font("Helvetica", Font.BOLD, 11));
        intervalLabel.setForeground(new Color(102, 102, 102));
        intervalLabel.setAlignmentX(Component.LEFT_ALIGNMENT);

        JPanel buttonFrame = new JPanel();
        buttonFrame.setLayout(new FlowLayout(FlowLayout.LEFT));
        buttonFrame.setBackground(new Color(240, 240, 240));

        int[] intervals = {15, 20, 30, 45, 60};
        for (int minutes : intervals) {
            JButton btn = createIntervalButton(minutes);
            buttonFrame.add(btn);

            if (minutes == 30) {
                activeIntervalBtn = btn;
            }
        }

        // Stats cards
        JPanel statsFrame = new JPanel();
        statsFrame.setLayout(new GridLayout(1, 2, 10, 0));
        statsFrame.setBackground(new Color(240, 240, 240));

        JPanel timeCard = createStatCard("Next drink in", "30", "minutes", new Color(102, 126, 234));
        timeLeftLabel = (JLabel) ((JPanel) timeCard.getComponent(1)).getComponent(0);

        JPanel drinksCard = createStatCard("Times hydrated", "0", "today", new Color(38, 208, 206));
        drinksLabel = (JLabel) ((JPanel) drinksCard.getComponent(1)).getComponent(0);

        statsFrame.add(timeCard);
        statsFrame.add(drinksCard);

        // Test button
        JButton testBtn = new JButton("🧪 Test Reminder (Click here)");
        testBtn.setFont(new Font("Helvetica", Font.BOLD, 12));
        testBtn.setBackground(new Color(102, 126, 234));
        testBtn.setForeground(Color.WHITE);
        testBtn.setBorder(BorderFactory.createEmptyBorder(12, 20, 12, 20));
        testBtn.setFocusPainted(false);
        testBtn.setMaximumSize(new Dimension(Integer.MAX_VALUE, 50));
        testBtn.addActionListener(e -> showReminder());

        // Timer status
        timerStatusLabel = new JLabel("⏱ Timer active - reminder in 30 minutes");
        timerStatusLabel.setFont(new Font("Helvetica", Font.PLAIN, 10));
        timerStatusLabel.setForeground(new Color(102, 102, 102));
        timerStatusLabel.setAlignmentX(Component.CENTER_ALIGNMENT);

        // Add components to content
        content.add(intervalLabel);
        content.add(Box.createVerticalStrut(8));
        content.add(buttonFrame);
        content.add(Box.createVerticalStrut(20));
        content.add(statsFrame);
        content.add(Box.createVerticalStrut(20));
        content.add(testBtn);
        content.add(Box.createVerticalStrut(10));
        content.add(timerStatusLabel);

        return content;
    }

    private JButton createIntervalButton(int minutes) {
        JButton btn = new JButton(minutes + "m");
        btn.setFont(new Font("Helvetica", Font.BOLD, 10));
        btn.setBackground(minutes == 30 ? new Color(102, 126, 234) : Color.WHITE);
        btn.setForeground(minutes == 30 ? Color.WHITE : new Color(51, 51, 51));
        btn.setBorder(BorderFactory.createEmptyBorder(8, 16, 8, 16));
        btn.setFocusPainted(false);
        btn.setPreferredSize(new Dimension(70, 35));

        btn.addActionListener(e -> setInterval(minutes));

        return btn;
    }

    private JPanel createStatCard(String label, String value, String unit, Color accentColor) {
        JPanel card = new JPanel();
        card.setLayout(new BorderLayout());
        card.setBackground(new Color(249, 249, 249));
        card.setBorder(BorderFactory.createLineBorder(new Color(224, 224, 224)));

        JPanel content = new JPanel();
        content.setLayout(new BoxLayout(content, BoxLayout.Y_AXIS));
        content.setBackground(new Color(249, 249, 249));
        content.setBorder(new EmptyBorder(10, 10, 10, 10));

        JLabel labelText = new JLabel(label);
        labelText.setFont(new Font("Helvetica", Font.PLAIN, 10));
        labelText.setForeground(new Color(153, 153, 153));
        labelText.setAlignmentX(Component.CENTER_ALIGNMENT);

        JLabel valueText = new JLabel(value);
        valueText.setFont(new Font("Helvetica", Font.BOLD, 32));
        valueText.setForeground(accentColor);
        valueText.setAlignmentX(Component.CENTER_ALIGNMENT);

        JLabel unitText = new JLabel(unit);
        unitText.setFont(new Font("Helvetica", Font.PLAIN, 10));
        unitText.setForeground(new Color(153, 153, 153));
        unitText.setAlignmentX(Component.CENTER_ALIGNMENT);

        content.add(labelText);
        content.add(Box.createVerticalStrut(5));
        content.add(valueText);
        content.add(Box.createVerticalStrut(4));
        content.add(unitText);

        card.add(content, BorderLayout.CENTER);
        return card;
    }

    private void setInterval(int minutes) {
        intervalMinutes = minutes;
        timeLeft = minutes;
        isSnoozed = false;

        // Update button colors
        for (Component comp : ((JPanel) ((JScrollPane) getContentPane()
                .getComponent(1)).getViewport().getComponent(0)).getComponents()) {
            // Would need to track buttons to update them
        }

        updateDisplay();
    }

    private void startTimer() {
        timerRunning = true;
        new Thread(() -> {
            while (timerRunning) {
                try {
                    Thread.sleep(60000); // 1 minute
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }

                timeLeft--;

                if (timeLeft <= 0) {
                    showReminder();
                    timeLeft = intervalMinutes;
                }

                SwingUtilities.invokeLater(this::updateDisplay);
            }
        }).start();
    }

    private void updateDisplay() {
        timeLeftLabel.setText(String.valueOf(timeLeft));
        drinksLabel.setText(String.valueOf(drinksCount));

        String status;
        if (isSnoozed) {
            status = String.format("⏱ Snoozed - reminder in %d minute%s", timeLeft, timeLeft != 1 ? "s" : "");
        } else {
            status = String.format("⏱ Timer active - reminder in %d minute%s", timeLeft, timeLeft != 1 ? "s" : "");
        }
        timerStatusLabel.setText(status);
    }

    private void showReminder() {
        SwingUtilities.invokeLater(() -> {
            JWindow reminderWindow = new JWindow();
            reminderWindow.setAlwaysOnTop(true);

            JPanel contentPanel = new JPanel();
            contentPanel.setLayout(new BoxLayout(contentPanel, BoxLayout.Y_AXIS));
            contentPanel.setBackground(Color.WHITE);
            contentPanel.setBorder(new EmptyBorder(30, 20, 30, 20));

            // Emoji label
            JLabel emojiLabel = new JLabel(walkingFrames[0]);
            emojiLabel.setFont(new Font("Dialog", Font.PLAIN, 100));
            emojiLabel.setAlignmentX(Component.CENTER_ALIGNMENT);

            // Title
            JLabel titleLabel = new JLabel("Time to hydrate!");
            titleLabel.setFont(new Font("Helvetica", Font.BOLD, 24));
            titleLabel.setForeground(new Color(26, 26, 26));
            titleLabel.setAlignmentX(Component.CENTER_ALIGNMENT);

            // Message
            JLabel messageLabel = new JLabel("<html><center>Your brain works better when you're hydrated.<br>Take a sip of water and keep coding! 🚀</center></html>");
            messageLabel.setFont(new Font("Helvetica", Font.PLAIN, 13));
            messageLabel.setForeground(new Color(102, 102, 102));
            messageLabel.setAlignmentX(Component.CENTER_ALIGNMENT);

            // Button panel
            JPanel buttonPanel = new JPanel();
            buttonPanel.setLayout(new GridLayout(1, 2, 8, 0));
            buttonPanel.setBackground(Color.WHITE);

            JButton drinkBtn = new JButton("✓ Yes, let me drink");
            drinkBtn.setFont(new Font("Helvetica", Font.BOLD, 12));
            drinkBtn.setBackground(new Color(38, 208, 206));
            drinkBtn.setForeground(Color.WHITE);
            drinkBtn.setFocusPainted(false);
            drinkBtn.setPadding(new Insets(12, 20, 12, 20));
            drinkBtn.addActionListener(e -> {
                handleDrink();
                reminderWindow.dispose();
            });

            JButton snoozeBtn = new JButton("⏱ Snooze 5m");
            snoozeBtn.setFont(new Font("Helvetica", Font.BOLD, 12));
            snoozeBtn.setBackground(new Color(240, 240, 240));
            snoozeBtn.setForeground(new Color(51, 51, 51));
            snoozeBtn.setFocusPainted(false);
            snoozeBtn.setPadding(new Insets(12, 20, 12, 20));
            snoozeBtn.addActionListener(e -> {
                handleSnooze();
                reminderWindow.dispose();
            });

            buttonPanel.add(drinkBtn);
            buttonPanel.add(snoozeBtn);

            // Add components
            contentPanel.add(emojiLabel);
            contentPanel.add(Box.createVerticalStrut(10));
            contentPanel.add(titleLabel);
            contentPanel.add(Box.createVerticalStrut(10));
            contentPanel.add(messageLabel);
            contentPanel.add(Box.createVerticalStrut(20));
            contentPanel.add(buttonPanel);

            reminderWindow.setContentPane(contentPanel);
            reminderWindow.setSize(450, 500);

            // Center on screen
            Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
            reminderWindow.setLocation(
                    (screenSize.width - 450) / 2,
                    (screenSize.height - 500) / 2
            );

            // Animation
            animateWindow(reminderWindow);

            reminderWindow.setVisible(true);
        });
    }

    private void animateWindow(JWindow window) {
        new Thread(() -> {
            int startY = window.getY();
            int targetY = window.getY();

            for (int i = 0; i < 20; i++) {
                try {
                    Thread.sleep(50);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }

                frameIndex = (frameIndex + 1) % walkingFrames.length;
                final int currentFrameIndex = frameIndex;

                // Update emoji in the window
                SwingUtilities.invokeLater(() -> {
                    // Would need to keep reference to emoji label
                });

                window.repaint();
            }
        }).start();
    }

    private void handleDrink() {
        drinksCount++;
        timeLeft = intervalMinutes;
        isSnoozed = false;
        updateDisplay();
    }

    private void handleSnooze() {
        timeLeft = 5;
        isSnoozed = true;
        updateDisplay();
    }

    public static void main(String[] args) {
        SwingUtilities.invokeLater(HydrationBuddy::new);
    }
}

File: drink.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>
#include <unistd.h>
#include <signal.h>
#include <pthread.h>

// ANSI Color codes
#define RESET   "\x1b[0m"
#define BOLD    "\x1b[1m"
#define BLUE    "\x1b[34m"
#define CYAN    "\x1b[36m"
#define GREEN   "\x1b[32m"
#define YELLOW  "\x1b[33m"
#define BG_BLUE "\x1b[44m"
#define BG_CYAN "\x1b[46m"

typedef struct {
    int interval_minutes;
    int time_left;
    int drinks_count;
    int is_snoozed;
    int timer_running;
    int show_menu;
} HydrationState;

// Walking animation frames of the app
const char *walking_frames[] = {"🚶", "🧘", "🏃", "🧘"};
int frame_count = 4;

void clear_screen() {
    system("clear");
}

void print_header() {
    printf("%s%s", BG_BLUE, RESET);
    printf("\n");
    printf("%s    💧 Hydration Buddy - Stay hydrated while you code!%s\n", BOLD BLUE, RESET);
    printf("\n");
}

void print_separator() {
    printf("%s━━━━━━━━━━━━━━━━━━━━%s\n", CYAN, RESET);
}

void print_main_menu(HydrationState *state) {
    clear_screen();
    print_header();
    
    printf("\n%s📊 Current Status:%s\n", BOLD, RESET);
    print_separator();
    
    printf("%s⏱  Time until next reminder:%s %s%d%s minutes\n", 
           BOLD, RESET, BOLD BLUE, state->time_left, RESET);
    printf("%s💧 Times hydrated today:%s %s%d%s\n\n", 
           BOLD, RESET, BOLD GREEN, state->drinks_count, RESET);
    
    if (state->is_snoozed) {
        printf("%s✓ Status:%s Snoozed (reminder in %d minute%s)\n\n", 
               BOLD, RESET, state->time_left, state->time_left != 1 ? "s" : "");
    } else {
        printf("%s✓ Status:%s Timer active (reminder in %d minute%s)\n\n", 
               BOLD, RESET, state->time_left, state->time_left != 1 ? "s" : "");
    }
    
    printf("%s🔔 Reminder Interval:%s\n", BOLD, RESET);
    printf("  ");
    int intervals[] = {15, 20, 30, 45, 60};
    for (int i = 0; i < 5; i++) {
        if (intervals[i] == state->interval_minutes) {
            printf("%s[%d] %s ", BOLD GREEN, intervals[i], RESET);
        } else {
            printf("[%d] ", intervals[i]);
        }
    }
    printf("\n\n");
    
    printf("%s📋 Menu:%s\n", BOLD, RESET);
    print_separator();
    printf("  %s1%s - Set interval to 15 minutes\n", BOLD CYAN, RESET);
    printf("  %s2%s - Set interval to 20 minutes\n", BOLD CYAN, RESET);
    printf("  %s3%s - Set interval to 30 minutes (default)\n", BOLD CYAN, RESET);
    printf("  %s4%s - Set interval to 45 minutes\n", BOLD CYAN, RESET);
    printf("  %s5%s - Set interval to 60 minutes\n", BOLD CYAN, RESET);
    printf("  %s6%s - Show reminder now (test)\n", BOLD CYAN, RESET);
    printf("  %s7%s - Drink water (reset timer)\n", BOLD CYAN, RESET);
    printf("  %s8%s - Snooze for 5 minutes\n", BOLD CYAN, RESET);
    printf("  %s0%s - Exit\n", BOLD CYAN, RESET);
    printf("\n%sEnter your choice: %s", BOLD, RESET);
}

void show_reminder(HydrationState *state) {
    clear_screen();
    print_header();
    
    printf("\n");
    printf("%s%s%s%s%s\n", BOLD, "                        ", walking_frames[0], RESET, "\n");
    printf("\n");
    printf("%s%s💧 Time to hydrate! 💧%s\n\n", BOLD BLUE, "                    ", RESET);
    printf("%s%s%s\n", CYAN, "    Your brain works better when you're hydrated.", RESET);
    printf("%s%s%s\n", CYAN, "    Take a sip of water and keep coding! 🚀", RESET);
    printf("\n");
    print_separator();
    printf("\n%s1%s - Yes, I drank water ✓\n", BOLD GREEN, RESET);
    printf("%s2%s - Snooze for 5 minutes ⏱\n", BOLD YELLOW, RESET);
    printf("\nEnter your choice: ");
}

void handle_reminder(HydrationState *state) {
    int choice;
    show_reminder(state);
    scanf("%d", &choice);
    
    if (choice == 1) {
        state->drinks_count++;
        state->time_left = state->interval_minutes;
        state->is_snoozed = 0;
        printf("\n%s✓ Great! Keep up the hydration!%s\n", GREEN BOLD, RESET);
        sleep(2);
    } else if (choice == 2) {
        state->time_left = 5;
        state->is_snoozed = 1;
        printf("\n%s⏱ Snoozed for 5 minutes. Next reminder soon!%s\n", YELLOW BOLD, RESET);
        sleep(2);
    }
}

void set_interval(HydrationState *state, int minutes) {
    state->interval_minutes = minutes;
    state->time_left = minutes;
    state->is_snoozed = 0;
    printf("%s✓ Reminder interval set to %d minutes%s\n", GREEN BOLD, minutes, RESET);
    sleep(1);
}

void *timer_thread(void *arg) {
    HydrationState *state = (HydrationState *)arg;
    
    while (state->timer_running) {
        sleep(60);  // 1 minute
        
        if (state->time_left > 0) {
            state->time_left--;
        }
        
        if (state->time_left == 0 && state->timer_running) {
            state->show_menu = 0;  // Signal to show reminder
            handle_reminder(state);
            state->show_menu = 1;  // Resume menu
        }
    }
    
    return NULL;
}

int main() {
    HydrationState state = {
        .interval_minutes = 30,
        .time_left = 30,
        .drinks_count = 0,
        .is_snoozed = 0,
        .timer_running = 1,
        .show_menu = 1
    };
    
    pthread_t timer_tid;
    pthread_create(&timer_tid, NULL, timer_thread, &state);
    
    int choice;
    
    while (state.timer_running) {
        if (state.show_menu) {
            print_main_menu(&state);
        }
        
        if (scanf("%d", &choice) != 1) {
            scanf("%*c");  // Clear invalid input
            continue;
        }
        
        switch (choice) {
            case 1:
                set_interval(&state, 15);
                break;
            case 2:
                set_interval(&state, 20);
                break;
            case 3:
                set_interval(&state, 30);
                break;
            case 4:
                set_interval(&state, 45);
                break;
            case 5:
                set_interval(&state, 60);
                break;
            case 6:
                handle_reminder(&state);
                break;
            case 7:
                state.drinks_count++;
                state.time_left = state.interval_minutes;
                state.is_snoozed = 0;
                printf("%s✓ Water intake recorded!%s\n", GREEN BOLD, RESET);
                sleep(1);
                break;
            case 8:
                state.time_left = 5;
                state.is_snoozed = 1;
                printf("%s⏱ Snoozed for 5 minutes%s\n", YELLOW BOLD, RESET);
                sleep(1);
                break;
            case 0:
                state.timer_running = 0;
                printf("\n%s👋 Stay hydrated! Goodbye!%s\n\n", BOLD CYAN, RESET);
                break;
            default:
                printf("%s✗ Invalid choice. Please try again.%s\n", BOLD, RESET);
                sleep(1);
        }
    }
    
    pthread_join(timer_tid, NULL);
    
    return 0;
}
